hamocc/mo_param_bgc.F90

```diff
@@ -43,73 +43,65 @@ module mo_param_bgc
 !
 !******************************************************************************
 
-  use mo_carbch,      only: atm,atm_co2,atm_n2,atm_o2,dmspar 
+  use mo_carbch,      only: atm,atm_co2,atm_n2,atm_o2,dmspar
   use mo_biomod,      only: atten_c,atten_f,atten_uv,atten_w,bkopal,bkphy,bkopal,bkzoo,bluefix,ctochl,dremn2o,dremopal,            &
                           & drempoc,dremsul,dyphy,ecan,epsher,fesoly,fetune,gammap,gammaz,grami,grazra,perc_diron,phytomi,         &
                           & pi_alpha,rcalc,rcar, rdn2o1,rdn2o2,rdnit0,rdnit1,rdnit2,relaxfe,remido,riron,rnit,rnoi,ro2ut,          &
-                          & ropal,spemor,tf0,tf1,tf2,tff,wcal,wdust,wopal,wpoc,zinges 
+                          & ropal,spemor,tf0,tf1,tf2,tff,wcal,wdust,wopal,wpoc,zinges
   use mo_sedmnt,      only: claydens,o2ut,rno3
   use mo_control_bgc, only: io_stdo_bgc,bgc_namelist
+  use mo_control_bgc, only: use_AGG, use_natDIC, use_BROMO, use_cisonew, use_WLIN
   use mo_param1_bgc,  only: iatmco2,iatmnco2,iatmo2,iatmn2,iatmc13,iatmc14,iatmbromo
   use mod_xc,         only: mnproc
-
-#ifdef AGG
+  ! agg
   use mo_biomod,      only: alar1,alar2,alar3,alow1,alow2,alow3,calmax,cellmass,cellsink,dustd1,dustd2,dustd3,dustsink,            &
                           & fractdim,fse,fsh,nmldmin,plower,pupper,safe,sinkexp,stick,tmfac,tsfac,vsmall,zdis
-#elif defined(WLIN)
+  ! wlin
   use mo_biomod,      only: wmin,wmax,wlin
-#endif
-#ifdef BROMO
+  ! bromo
   use mo_biomod,      only: rbro
   use mo_carbch,      only: atm_bromo,fbro1,fbro2
-#endif
-#ifdef cisonew
+  ! cisonew
   use mo_biomod,      only: bifr13,bifr14,c14fac,prei13,prei14,re1312,re14to
   use mo_carbch,      only: atm_c13, atm_c14,c14_t_half,c14dec
-#endif
-#ifdef natDIC
+  ! natDIC
   use mo_carbch,      only: atm_co2_nat
-#endif
 
   implicit none
 
   private
-   
+
   public :: ini_parambgc
 
   ! Module-wide parameters (used in more than one subroutine)
-#ifndef AGG
-  REAL :: dustd1, dustd2, dustsink
-#endif
-#ifdef cisonew 
-  REAL :: beta13, alpha14, d14cat, d13c_atm  
-#endif
+  REAL :: dustd1, dustd2, dustsink !AGG
```

Comment:
I am running into compilation problems - the line with dustd1, dustd2, dustsink needs to be deleted.

Reply:
I don't understand this comment. I have no problems compiling.

Reply:
Hi @mvertens , `dustd1, dustd2, dustsink` are already included by `use mo_biomod, only:` above so that at least on my side, the compiler complains `Symbol ‘dustd1’ at (1) conflicts with symbol from module ‘mo_biomod’, use-associated at (2)` (compiled locally via meson). Similarly, `dustd2` and `dustsink` are also already defined. As written, it was very twisted before (these were only included via `use` when `AGG` was switched on, while then the local variables were required inside the routine for some initialization). Deletion of the line 
```
  REAL :: dustd1, dustd2, dustsink !AGG
```
should do the job.

Reply:
Yes - you are right. Sorry for not seeing that.

Reply:
No worries - I am happy to be able to help :-)

Reply:
I've pushed the fix back.